test/sharness/t0271-filestore-utils.sh

```diff
@@ -0,0 +1,195 @@
+		ipfs filestore ls $FILE1_HASH > ls_actual &&
+		grep -q somedir/file1 ls_actual
+	'
+
+	test_expect_success "can retrieve multi-block file" '
+		ipfs cat $FILE3_HASH > file3.data &&
+		test_cmp somedir/file3 file3.data
+	'
+}
+
+test_filestore_verify() {
+	test_expect_success "ipfs filestore verify' output looks good'" '
+		ipfs filestore verify | LC_ALL=C sort > verify_actual
+		test_cmp verify_expect verify_actual
+	'
+
+	test_expect_success "'ipfs filestore verify HASH' works" '
+		ipfs filestore verify $FILE1_HASH > verify_actual &&
+		grep -q somedir/file1 verify_actual
+	'
+
+	test_expect_success "rename a file" '
+		mv somedir/file1 somedir/file1.bk
+	'
+
+	test_expect_success "can not retrieve block after backing file moved" '
+		test_must_fail ipfs cat $FILE1_HASH
+	'
+
+	test_expect_success "'ipfs filestore verify' shows file as missing" '
+		ipfs filestore verify > verify_actual &&
+		grep no-file verify_actual | grep -q somedir/file1
+	'
+
+	test_expect_success "move file back" '
+		mv somedir/file1.bk somedir/file1
+	'
+
+	test_expect_success "block okay now" '
+		ipfs cat $FILE1_HASH > file1.data &&
+		test_cmp somedir/file1 file1.data
+	'
+
+	test_expect_success "change first bit of file" '
+		dd if=/dev/zero of=somedir/file3 bs=1024 count=1
+	'
+
+	test_expect_success "can not retrieve block after backing file changed" '
+		test_must_fail ipfs cat $FILE3_HASH
+	'
+
+	test_expect_success "'ipfs filestore verify' shows file as changed" '
+		ipfs filestore verify > verify_actual &&
+		grep changed verify_actual | grep -q somedir/file3
+	'
+}
+
+cat <<EOF > dups_expect
+$FILE1_HASH
+EOF
+
+test_filestore_dups_rm() {
+	test_expect_success "'ipfs filestore dups'" '
+		ipfs add --raw-leaves somedir/file1 &&
+		ipfs filestore dups > dups_actual &&
+		test_cmp dups_expect dups_actual
+	'
+
+	test_expect_success "remove non-filestore block of dup ok" '
+		ipfs filestore rm --non-filestore $FILE1_HASH &&
+		ipfs filestore dups > dups_actual &&
+		test_cmp /dev/null dups_actual
+	'
+
+	test_expect_success "block still in filestore" '
+		ipfs filestore ls $FILE1_HASH | grep -q file1
+	'
+
+	test_expect_success "remove non-duplicate pinned block not ok" '
+		test_must_fail ipfs filestore rm $FILE1_HASH 2> rm_err &&
+		grep -q pinned rm_err
+	'
+
+	test_expect_success "remove filestore block of dup ok" '
+		ipfs add --raw-leaves somedir/file1 &&
+		ipfs filestore rm $FILE1_HASH &&
+		ipfs filestore dups > dups_actual &&
+		test_cmp /dev/null dups_actual
+	'
+}
+
+init_ipfs_filestore() {
+	test_expect_success "clean up old node" '
+		rm -rf "$IPFS_PATH" mountdir ipfs ipns
+	'
+
+	test_init_ipfs
+
+	test_expect_success "enable filestore config setting" '
+		ipfs config --json Experimental.FilestoreEnabled true
+	'
+}
+
+test_init_dataset
+
+init_ipfs_filestore
+
+test_filestore_adds
+
+test_filestore_verify
+
+test_filestore_dups_rm
+
+echo "WORKING DIR"
+echo "IPFS PATH = " $IPFS_PATH
+pwd
+
+
```

Comment:
It does as it makes harder to understand the tests, which means that in future, when they will have to be changed, already high chance of mistake will be even higher.